adage/genes/views.py

```diff
@@ -1,3 +1,57 @@
-from django.shortcuts import render
+from rest_framework.viewsets import ReadOnlyModelViewSet
+from .models import Gene
+from .serializers import GeneSerializer
 
-# Create your views here.
+
+class GeneViewSet(ReadOnlyModelViewSet):
+    """Genes viewset"""
+
+    http_method_names = ['get']
+    serializer_class = GeneSerializer
+
+    def get_queryset(self):
+        queryset = Gene.objects.all()
+
+        # 'search' parameter, which does full text search based on the
+        # following four fields in Gene model:
+        # - "standard_name": highest priority (A: 1.0);
+        # - "systematic_name": second highest priority (B: 0.4);
+        # - "description" and "crossref__xrid": lowest priority (C: 0.2).
+        search_str = self.request.query_params.get('search', None)
+        if search_str is not None:
+            from django.contrib.postgres.search import (
+                SearchQuery, SearchRank, SearchVector
+            )
+
+            standard_vector = SearchVector('standard_name', weight='A')
+            systematic_vector = SearchVector('systematic_name', weight='B')
+            other_vector = SearchVector(
+                'description', 'crossref__xrid', weight='C'
+            )
+
+            vectors = standard_vector + systematic_vector + other_vector
+            query = SearchQuery(search_str)
+            queryset = queryset.annotate(
+                rank=SearchRank(vectors, query)
+            ).filter(rank__gte=0.1
+            ).order_by('-rank', 'standard_name')
```

Comment:
I need to spend more time with the Postgres search documentation but it looks like this works.

Hmm. I thought that Postgres could do stemming on search terms, but that doesn't appear to be happening in this case. For example, I searched on:

> https://py3-adage.greenelab.com/api/v1/genes/?search=transport

and got some results. Then I searched on:

> https://py3-adage.greenelab.com/api/v1/genes/?search=transporter

and got different results. If stemming is working, I would expect those two search results to at least have some overlap.

Reply:
Is there a parameter or configuration you need to set to make stemming work?

Reply:
It turned out that I forgot to add `config='english'` when building search vectors and search query. After I deploy it on the test server, you can compare the search results again.

Reply:
This rather nicely underscores my point about trying to change the configuration centrally. Then you wouldn't have to remember to override it for each index.

Reply:
Re-reading what I wrote just now and that last comment came across as more snide than intended. I'm a bit frustrated that I can't suggest more constructive ideas than "this doesn't look right" because I'm not so familiar with Postgres full text search. I'll do some reading up and will hopefully have some better suggestions next time around.